lib/batch_loader/executor_proxy.rb

```diff
@@ -9,7 +9,7 @@ class ExecutorProxy
     def initialize(default_value, key, &block)
       @default_value = default_value
       @block = block
-      @block_hash_key = "#{key}#{block.source_location}"
+      @block_hash_key = [block.source_location, key]
```

Comment:
I decided to reverse `key` and `block.source_location`, since `key` is used to subcategorize.